### ISSUE TITLE
Orientation incorrect when set to .portraitUpsideDown on iPhone

### ISSUE THREAD

**Issue by cerupcat** [huri000/SwiftEntryKit#143]
I have an iPhone app that supports UIInterfaceOrientationMask.portrait and UIInterfaceOrientationMask.portraitUpsideDown (no landscape). When orientation is `.portraitUpsideDown` and I present a SwiftEntryKit, it is displayed upside down (as if the phone was in portrait and not upside down). I've tried setting the `isRotationEnabled` to false (and true), but it still occurs. My guess is the SwiftEntryKit `EKRootViewController` needs to have the `supportedInterfaceOrientations` set on it so it knows what orientations to support, but it doesn't appear its possible to extend the `EKRootViewController` in order to do so.

Here's the code I'm using:

``` 
var attributes = EKAttributes.centerFloat
    attributes.entryBackground = .color(color: backgroundColor)
    attributes.shadow = .active(with: .init(color: .black, opacity: 0.3, radius: 10, offset: .zero))
    attributes.entranceAnimation = .init(scale: .init(from: 0.9, to: 1, duration: 0.4, spring: .init(damping: 1, initialVelocity: 0)), fade: .init(from: 0, to: 1, duration: 0.3))
    attributes.exitAnimation = .init(fade: .init(from: 1, to: 0, duration: 0.2))
    attributes.positionConstraints.maxSize = .init(width: .constant(value: UIScreen.main.bounds.width), height: .intrinsic)
    attributes.positionConstraints.isRotationEnabled = false

    let title = EKProperty.LabelContent(text: text, style: .init(font: UIFont.systemFont(ofSize: 16, weight: .regular), color: textColor, alignment: .center))
    let description = EKProperty.LabelContent(text: detailText, style: .init(font: UIFont.systemFont(ofSize: 14, weight: .bold), color: textColor, alignment: .center))

    let simpleMessage: EKSimpleMessage
    if let image = image {
        let image = EKProperty.ImageContent(image: image, size: CGSize(width: 35, height: 35))
        simpleMessage = EKSimpleMessage(image: image, title: title, description: description)
    } else {
        simpleMessage = EKSimpleMessage(image: nil, title: title, description: description)
    }

    let notificationMessage = EKNotificationMessage(simpleMessage: simpleMessage)

    let contentView = EKNotificationMessageView(with: notificationMessage)

    SwiftEntryKit.display(entry: contentView, using: attributes)
```

![simulator screen shot - iphone 6 plus - 2018-12-18 at 11 23 24](https://user-images.githubusercontent.com/204151/50304174-13eb8480-0444-11e9-85be-677eb9056840.png)

**Comment by cerupcat** [comment]
Thanks @huri000.

As a test, I added the following to 'EKRootViewController' and everything worked as expected. I can try putting a `supportedInterfaceOrientations` variable on the `lastAttributes.positionConstraints` and submit a PR.

```
    override open var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        if UIDevice.current.userInterfaceIdiom == .pad {
            return .all
        } else{
            return [UIInterfaceOrientationMask.portrait, UIInterfaceOrientationMask.portraitUpsideDown]
        }
    }
```

**Comment by cerupcat** [comment]
I looked into making a PR where this would be configurable, but it crashes if I add a new `supportedInterfaceOrientations: UIInterfaceOrientationMask?` variable to `positionConstraints` so I'm not sure I understand the underlying framework enough to submit a PR.

**Comment by huri000** [comment]
I'll probably release a new version this weekend. 
I'm gonna add the following construct to PositionConstraints (which will replace the boolean that controls the rotation today): 
```Swift
/** Rotation related position constraints */
public struct Rotation {
    
    /** Attributes of supported interface orientations */
    public enum SupportedInterfaceOrientation {
        
        /** Uses standard supported interface orientation (target specification in general settings) */
        case standard
        
        /** Supports all orinetations */
        case all
        
        /** Supports specified orientationed */
        case specific(value: UIInterfaceOrientationMask)
    }
    
    /** Autorotate the entry along with the device orientation enabled / disabled */
    public var isEnabled = false
    
    /** The screen autorotates with accordance to this option */
    public var supportedInterfaceOrientations = SupportedInterfaceOrientation.all
    
    public init() {}
}
```
Let me know your thoughts on this.

**Comment by cerupcat** [comment]
Looks like that should work well.

**Comment by huri000** [comment]
Hi @cerupcat - The fix is available on version `0.8.8`. 🎉
Please check it out. Let me know if that works for you.
